File: agent/02-multi-agent/go/agent.go
// agent.go —— 单 Agent：name + role + 一次 LLM 调用。整文件 cp 进项目即可。
package main

import (
	"context"

	"github.com/sashabaranov/go-openai"
)

type Agent struct {
	Name        string
	Role        string  // system prompt
	Client      *openai.Client
	Model       string
	Temperature float32 // 默认 0.3
}

func (a *Agent) Execute(ctx context.Context, task, extra string) (string, error) {
	user := task
	if extra != "" {
		user = task + "\n\n上游产物：\n" + extra
	}
	temp := a.Temperature
	if temp == 0 {
		temp = 0.3
	}
	resp, err := a.Client.CreateChatCompletion(ctx, openai.ChatCompletionRequest{
		Model: a.Model,
		Messages: []openai.ChatCompletionMessage{
			{Role: openai.ChatMessageRoleSystem, Content: a.Role},
			{Role: openai.ChatMessageRoleUser, Content: user},
		},
		Temperature: temp,
		MaxTokens:   400,
	})
	if err != nil {
		return "", err
	}
	return resp.Choices[0].Message.Content, nil
}

File: agent/02-multi-agent/rust/src/agent.rs
//! agent.rs —— 单 Agent。整文件 cp 进项目即可。

use serde_json::{json, Value};

#[derive(Clone)]
pub struct Config {
    pub base_url: String,
    pub api_key: String,
    pub model: String,
}

pub struct Agent {
    pub name: String,
    pub role: String,
    pub cfg: Config,
    pub temperature: f32, // 默认 0.3
}

impl Agent {
    pub fn execute(&self, task: &str, extra: &str) -> Result<String, String> {
        let user = if extra.is_empty() {
            task.to_string()
        } else {
            format!("{task}\n\n上游产物：\n{extra}")
        };
        let temp = if self.temperature == 0.0 { 0.3 } else { self.temperature };
        let resp: Value = ureq::post(&format!("{}/chat/completions", self.cfg.base_url))
            .set("Authorization", &format!("Bearer {}", self.cfg.api_key))
            .set("Content-Type", "application/json")
            .timeout(std::time::Duration::from_secs(60))
            .send_json(json!({
                "model": self.cfg.model,
                "messages": [
                    {"role": "system", "content": self.role},
                    {"role": "user", "content": user},
                ],
                "temperature": temp,
                "max_tokens": 400,
            }))
            .map_err(|e| e.to_string())?
            .into_json::<Value>()
            .map_err(|e| e.to_string())?;
        Ok(resp["choices"][0]["message"]["content"].as_str().unwrap_or("").to_string())
    }
}
